zerver/lib/request.py

```diff
@@ -400,6 +400,10 @@ def _wrapped_req_func(
                 post_var_names += param.aliases
                 post_var_name = None
 
+                patch_body: Optional[QueryDict] = None
+                if request.method == "PATCH":
+                    patch_body = QueryDict(request.body)
```

Comment:
I don't know how PATCH bodies are supposed to work per RFCs, but this appears to be correct if we're supposed to parse the body of the request as JSON. 

I'd kinda have expected Django to parse the body of the request via `request.POST` for us, though.

Reply:
At least in the `/json/settings` endpoint, we don't PATCH per RFC at all: we're using the verb, but not the RFC semantics.

- For the whole, dry, deep dive: https://www.rfc-editor.org/rfc/rfc5789
- For a more opinionated and skimmable take on endpoints that use PATCH the way our `/json/settings` does: https://williamdurand.fr/2014/02/14/please-dont-patch-like-that/

Since the body of a PATCH is inherently different in nature than the body of a POST, Django doesn't reuse `request.POST` to store the data. I did, however, just find `request.data` which might save the explicit `QueryDict` creation. I'll play around with it and see if it makes anything easier/clearer: https://www.django-rest-framework.org/api-guide/requests/#data

Reply:
`request.data` isn't in our version of Django, so disregard that last sentence or two in the above comment. The PR as proposed (with explicit `QueryDict` creation) is, so far, my best idea to solve this.